### ISSUE TITLE
Header validation tripping on known safe emails due to References header

### ISSUE THREAD

**Issue by garrettlr** [bbottema/simple-java-mail#171]
Hey @bbottema , 

I've had this come up a few times now and I'm not sure exactly why or how to avoid it. 

for a recent example see below: 

```org.simplejavamail.mailer.MailerException: Suspected of injection attack, field: email.header.References with suspicious value: <DOKOGW56MP@zendesk.com>
<DOKOGW56MP_5bbd3cc7769cf_c3c93fc52d8cb9781275fb_sprut@zendesk.com>
```

for background, I've used simple-java-mail to power an intranet site which only replies to emails we've received from third parties we work with, primarily to centralize identity, so header validation is of minor concern to us.

is it possible to just fail quietly?

thanks!

**Comment by bbottema** [comment]
Hi @garrettlr, can you provide a stacktrace or else which validation fails in the code?

**Comment by bbottema** [comment]
@garrettlr, I can't proceed with this until you provide some more info. I would like to understand better what is going on here and preferably be able to reproduce it on my own laptop.

**Comment by madalinl** [comment]
Hi @bbottema, maybe I can help you with some info.
I've stumbled on this error trying to reply to an valid e-mail:

`org.simplejavamail.mailer.MailerException: Suspected of injection attack, field: email.header.References with suspicious value: <aaabbbccc@xxxyyyzzz.eurprd07.prod.outlook.com>
<dddeeefff@xxxyyyzzz.eurprd07.prod.outlook.com>
        at org.simplejavamail.mailer.Mailer.scanForInjectionAttack(Mailer.java:337) ~[simple-java-mail-5.1.3.jar!/:na]
        at org.simplejavamail.mailer.Mailer.validate(Mailer.java:305) ~[simple-java-mail-5.1.3.jar!/:na]
        at org.simplejavamail.mailer.Mailer.sendMail(Mailer.java:238) ~[simple-java-mail-5.1.3.jar!/:na]
        at org.simplejavamail.mailer.Mailer.sendMail(Mailer.java:230) ~[simple-java-mail-5.1.3.jar!/:na]
`
The References header contains 2 values separated by '\n\r'.

Library version is 5.0.3

Hope this helps

**Comment by bbottema** [comment]
`\r\n` characters are [attack vectors for header injections](https://portswigger.net/kb/issues/00200800_smtp-header-injection) and is exactly what triggers the exception. Can you explain a little bit more about why this would be a valid use case?

**Comment by ScottPeterJohnson** [comment]
This happened to me when using the `EmailBuilder.replyingTo()` construct on a message that already contains a `References` header. JavaMail builds a new `References` header with both values, which is "folded", and thus probably contains a newline.

**Comment by bbottema** [comment]
Thanks @ScottPeterJohnson, I'll see if I can reproduce the bug.

**Comment by bbottema** [comment]
Fix released in v5.1.7. Please verify!
